Pr template (#935)
o3de/o3de.org: .github/pull_request_template.md

<!--

>     Thanks for your contribution to the Open 3D Engine documentation! Before creating your pull
>     request, we ask you to sign off on our submission checklist to ensure that your PR comes through
>     quickly. Our reviewers' role is to make sure that your submission follows these best practices.

Reviewer: Wondering if we should add a caveat that this is intended for non-trivial pull requests. Something like:
```suggestion
    Thanks for your contribution to the Open 3D Engine documentation! Before creating your pull
    request, we ask you to sign off on our submission checklist to ensure that your PR comes through
    quickly. Our reviewers' role is to make sure that all non-trivial submissions follow these best practices.
```

>     By submitting your pull request with DCO singoff, you agree to the Code of Conduct and
>     Open 3D Engine documentation and website licensing terms.

Reviewer: While I always enjoy a good sing off...
```suggestion
    By submitting your pull request with DCO signoff, you agree to the Code of Conduct and
    Open 3D Engine documentation and website licensing terms.
```

-->

## Change summary

<!-- Provide a short description of your changes. -->

> 

Author: We may be able to do this with the new GitHub Issue Forms: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

For now, I think that we should have manual entry because I don't want to over-automate just to get basic things we need in.

### Submission Checklist:

* [ ] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [ ] **Answer the question at hand** - Does the documentation answer a *why*, *how*, or *where* type of question?

> * [ ] **Consistency** - Does the formatting of text and style of writing remain *relatively consistent*?

Reviewer: Not sure what the intent is here, maybe because of the ambiguity with the word "relatively". Do you mean "consistent relative to the rest of the topic or guide that it's in"? Or do you mean "more or less (or 'mostly') consistent"?

Author: ```suggestion
* [ ] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
```
